packages/ui/src/forum/components/Thread/NewThreadPost.tsx

```diff
@@ -18,45 +23,67 @@ type GetTransaction = (
 
 export interface NewPostProps {
   getTransaction: GetTransaction
+  replyTo?: ForumPost
+  removeReply: () => void
 }
 
-export const NewThreadPost = ({ getTransaction }: NewPostProps) => {
-  const [postText, setText] = useState('')
-  const [isEditable, setEditable] = useState(false)
-  const { active } = useMyMemberships()
-  const { showModal } = useModal()
+export const NewThreadPost = React.forwardRef(
+  ({ getTransaction, replyTo, removeReply }: NewPostProps, ref: React.ForwardedRef<HTMLDivElement>) => {
+    const [postText, setText] = useState('')
+    const [isEditable, setEditable] = useState(false)
+    const { active } = useMyMemberships()
+    const { showModal } = useModal()
 
-  if (!active) {
-    return <TextBig>Pick an active membership to post in this thread</TextBig>
-  }
+    if (!active) {
+      return <TextBig ref={ref}>Pick an active membership to post in this thread</TextBig>
+    }
 
-  return (
-    <RowGapBlock gap={8}>
-      <InputComponent inputSize="auto">
-        <EditorMemo setNewText={setText} />
-      </InputComponent>
-      <ButtonsGroup>
-        <ButtonPrimary
-          size="medium"
-          onClick={() => {
-            const transaction = getTransaction(postText, isEditable)
-            transaction &&
-              showModal<CreatePostModalCall>({
-                modal: 'CreatePost',
-                data: { postText, transaction, isEditable },
-              })
-          }}
-          disabled={postText === ''}
-        >
-          Post a reply
-        </ButtonPrimary>
-        <Checkbox id="set-editable" onChange={setEditable} isChecked={isEditable}>
-          Keep editable
-        </Checkbox>
-      </ButtonsGroup>
-    </RowGapBlock>
-  )
-}
+    return (
+      <RowGapBlock gap={8} ref={ref}>
+        {replyTo && (
+          <Reply>
+            <ReplyBadge>
+              <ButtonsGroup align="left">
+                <ArrowReplyIcon />{' '}
+                <Badge>
+                  <Link to={window.location.href}>Replies to {replyTo.author.handle}</Link>
+                </Badge>
+              </ButtonsGroup>
+              <ButtonsGroup align="right">
+                <ButtonPrimary size="small" onClick={removeReply}>
+                  <CrossIcon />
+                </ButtonPrimary>
+              </ButtonsGroup>
```

Comment:
Could you try to make the button be displayed on the right?

A `margin-left: auto` should [work](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Flexible_Box_Layout/Aligning_Items_in_a_Flex_Container#using_auto_margins_for_main_axis_alignment).